### PR TITLE
Account for msats in LND routing fees

### PR DESCRIPTION
**Pull request by deregs**
# Description

Relates to issue: **[569](https://github.com/ZeusLN/zeus/issues/569)**

Only showing decimal places if there are msats involved in the fee:
Screen Shot 2022-02-07 at 6 12 40 PM

This pull request is categorized as a:

- [x] New feature
- [ ] Bug fix
- [ ] Code refactor
- [ ] Configuration change
- [ ] Locales update
- [ ] Other

## Checklist
- [x] I’ve run `npm run tsc` and made sure my code compiles correctly
- [x] I’ve run `npm run lint` and made sure my code didn’t contain any problematic patterns
- [x] I’ve run `npm run prettier` and made sure my code is formatted correctly
- [x] I’ve run `npm run test` and made sure all of the tests pass

## Testing

If you added new functionality or fixed a bug, did you add new unit tests?

- [x] No, I’m a fool
- [ ] Yes
- [ ] N/A

I have tested this PR on the following platforms (please specify OS version and phone model/VM):

- [ ] Android
- [x] iOS 13.7 on iPhone 11 Pro

I have tested this PR with the following types of nodes (please specify node version and API version where appropriate):

- [x] LND
- [ ] c-lightning-REST
- [ ] spark
- [ ] Eclair
- [ ] LndHub

### Locales
- [ ] I’ve added new locale text that requires translations
- [ ] I’m aware that new translations should be made on the Zeus [Transfix page](https://www.transifex.com/ZeusLN/zeus/) and not directly to this repo

### Third Party Dependencies and Packages

- [ ] Contributors will need to run `npm install` after this PR is merged in
- [ ] 3rd party dependencies have been modified:
    * verify that `package.json` and `package-lock.json` have been properly updated
    * verify that dependencies are installed for both iOS and Android platforms

### Other:

- [ ] Changes were made that require an update to the README
- [ ] Changes were made that require an update to onboarding

### CODE REVIEW

File: stores/UnitsStore.ts [hunk 1]
```diff
@@ -62,8 +62,9 @@ export default class UnitsStore {
         const { fiat } = settings;
         const units = fixedUnits || this.units;
 
-        const [wholeSats] = value.toString().split('.');
-        const sats = (wholeSats && Number(wholeSats)) || 0;
+        // Rounding sats to 3 decimal places because there was a rounding error which
+        // caused repeated 9s from the 4th decimal place onwards in some situations
+        const sats = Math.round(Number(value) * 1000) / 1000;
```
Comment (kaloudis) [inline stores/UnitsStore.ts:67]: Might be time to bring in BigNumber https://mikemcl.github.io/bignumber.js/

Comment (deregs) [reply stores/UnitsStore.ts]: Cool, added BigNumber in FeeStore to prevent the rounding errors

File: package.json [hunk 2]
```diff
@@ -31,6 +31,7 @@
     "assert": "1.5.0",
     "base-64": "0.1.0",
     "bc-ur": "0.1.6",
+    "bignumber.js": "^9.0.2",
```
Comment (kaloudis) [inline package.json:34]: please lock the version number here by removing the caret. Also make sure `package-lock.json` gets committed

**Review by kaloudis — CHANGES_REQUESTED** [review]

**Comment by kaloudis** [comment]
haha you got a rebase from hell. check the diff.

**Comment by deregs** [comment]
> haha you got a rebase from hell. check the diff.

Yeah lol I messed up bad on that rebase so just opened a fresh PR

**Pull request closed without merging**